src/encoder.rs

```diff
@@ -3,33 +3,52 @@ use std::io;
 use crate::{AvifData, AvifImage};
 use libavif_sys as sys;
 
+/// AVIF image encoder
 pub struct Encoder {
     encoder: *mut sys::avifEncoder,
 }
 
 impl Encoder {
+    /// Create a new encoder with default settings
```

Comment:
Can we document default values?

Also, what if instead of panicking on out-of-range values (i.e., max_threads, quantizer), we just clamped to permitted values? Do you have an opinion?

Reply:
Looking into libavif's source they clamp the values, so I guess it makes sense to do the same here.